api/metadata.proto

```diff
@@ -31,7 +31,13 @@ option go_package = "go.thethings.network/lorawan-stack/pkg/ttnpb";
 message RxMetadata {
   option (gogoproto.populate) = false;
 
-  GatewayIdentifiers gateway_ids = 1 [(gogoproto.embed) = true, (gogoproto.nullable) = false, (validate.rules).message.required = true];
+  oneof source {
+    option (validate.required) = true;
+
+    GatewayIdentifiers gateway_ids = 1 [(gogoproto.customname) = "GatewayIDs"];
+    PacketBrokerMetadata packet_broker = 18;
+  }
```

Comment:
This violates our compatibility commitment. `RxMetadata` is included in `ApplicationUplink`, so it's external API. Previously the `gateway_ids` was indicated as non-nullable and required, now we may omit it, which may break applications that enforce the old validation rule (for packetbroker uplinks).

To be fully compliant, we can simply add the `PacketBrokerMetadata`, but also still include the `GatewayIdentifiers` as we do now (with the `packetbroker` ID, as we already did in The Things Enterprise Stack).

Reply:
Ah yeah the JSONPB ...